app/main.py

```diff
@@ -371,32 +399,179 @@ def update_live_alerts_data(n_intervals, ongoing_live_alerts, ongoing_frame_urls
+                try:
+                    # For each live alert, we fetch the URL of the associated frame
+                    img_url = api_client.get_media_url(row["media_id"]).json()["url"]
+
+                except Exception:
+                    # This is just a security in case we cannot retrieve the URL of the detection frame
+                    img_url = ''
+
+                # We now want to fill-in the dictionary that will contain the URLs of the detection frames
+                if str(row['event_id']) not in dict_images_url_live_alerts.keys():
+                    # This is a new event, so we need to instantiate the key / value pair
+                    dict_images_url_live_alerts[str(row['event_id'])] = [img_url]
+
+                else:
+                    # We already have some URLs for this event and we simply append the latest frame to the list of URLs
+                    dict_images_url_live_alerts[str(row['event_id'])].append(img_url)
+
+            # Merging yaw (azimuth) field from devices_data
+            all_devices = pd.DataFrame(devices_data)
+
+            # We restrict the DataFrame to useful information
+            devices_yaw = all_devices[['id', 'yaw']].copy()
+
+            # We merge it with the live_alerts DataFrame
+            live_alerts = pd.merge(
+                live_alerts, devices_yaw,
+                how='left',
+                left_on=['device_id'], right_on=['id']
+            )
+
+            # We drop the azimuth associated with the alert as we will focus on the yaw of the device
+            live_alerts = live_alerts.drop(['azimuth'], axis=1)
+
+            # We rename columns to avoid any ambibguity (id_y is the id of the device)
+            live_alerts.rename(columns={'id_x': 'id', 'id_y': 'd_id'}, inplace=True)
+
+            # We store the IDs of newly loaded alerts in a dedicated list
+            # This will serve as the source of truth to know what frame URLs have already been fetched or not
+            new_loaded_frames = list(live_alerts['id'].unique())
 
-        if row['event_id'] not in dict_images_url_live_alerts.keys():
-            dict_images_url_live_alerts[row['event_id']] = []
-            dict_images_url_live_alerts[row['event_id']].append(img_url)
+            # We convert the live_alerts DataFrame into a JSON that can be stored in a dcc.Store component
+            live_alerts = live_alerts.to_json(orient='records')
+
+            # Reminder: in this case, this is the first time that we load alert data
+
+            # So we update all outputs:
+            # - the storage component that contains alert data in JSON format;
+            # - the storage component that contains the dictionary with detection frame URLs;
+            # - the storage component that serves as source of truth for the list of already loaded alerts
+
+            return live_alerts, dict_images_url_live_alerts, {'loaded_frames': new_loaded_frames}, 5 * 1000
```

Comment:
Switching from 15 seconds to 5 seconds seems fair, any idea on how you decided to setup 15 seconds for the first -lazy- callback exec (seems safe as well though) ?

Reply:
Typically, with the live alerts that are currently in the DB, the first callback takes a bit more than 10 seconds if I am not mistaken. So I thought it was important to take some safety margin although I doubt we will have as many live alerts as right now once deployed on the field (difficult to anticipate actually because firemen will probably use the acknowledgement button but at the same time, we can have many frames following each other for the same event...). In any case, I feel like leaving some comfortable loading time for the first callback to run should not be too much of an issue as long as the following callbacks are shorter and executed more often!